prusti-viper/src/encoder/type_encoder.rs

```diff
@@ -755,34 +756,22 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
 }
 
 /// Compute the values that a discriminant can take.
-pub fn compute_discriminant_values(adt_def: &ty::AdtDef, tcx: ty::TyCtxt) -> Vec<i128> {
+pub fn compute_discriminant_values<'tcx>(adt_def: &'tcx ty::AdtDef, tcx: ty::TyCtxt<'tcx>) -> Vec<i128> {
     let mut discr_values: Vec<i128> = vec![];
     // Handle *signed* discriminats
```

Comment:
Is this comment still true? Are there unsigned discriminants; do we handle them elsewhere?

Reply:
Yes. I just added a test that fails if we don't call `sign_extend`.

Reply:
Ah, you meant why there is no code block to handle unsigned discriminants? I'm clarifying that.